jina/executors/indexers/dbms/keyvalue.py

```diff
@@ -0,0 +1,94 @@
+import pickle
+from typing import List, Optional
+
+from jina.drivers.dump import DumpPersistor, DumpTypes
+from jina.executors.indexers.dbms import BaseDBMSIndexer
+from jina.executors.indexers.keyvalue import BinaryPbIndexer
+
+
+class DBMSBinaryPbIndexer(BinaryPbIndexer, BaseDBMSIndexer):
```

Comment:
Will it be in our vision @maximilianwerk @cristianmtr @florian-hoenicke to split BinaryPbIndexer?

Right now with this construction, the separation is not fully there. As it is now, this class still has a `query` method. I would expect in the future this separation can be full

Reply:
This class also suffers from the Diamond Problem.

Reply:
No separation yet. Especially since this is not fully ready for production. But yes, it should be done at some point.

Yes, I realize this class is not fully "done" yet. Was going to ask if we have a pattern for adapting another Indexer's storage methods (BinaryPb's) into another one

Reply:
Refactor common code to a Mixin which is used by both classes might be a proper approach.

Reply:
yes, we can see it when this is in a more mature state , but we will need to remember to do so.